### PR TITLE
CURA-3730: Hide Top/Bottom and Inner Wall legends in "Material Color"

### PR DESCRIPTION
**Pull request by LipuFei**
Don't show "Top/Bottom" and "Inner Wall" legends in "Material Color" in Layer View.

### CODE REVIEW

File: plugins/LayerView/LayerView.qml [hunk 1]
```diff
@@ -127,6 +127,16 @@ Item
...
+
+                    typesLegenModelNoCheck.clear();
+                    typesLegenModelNoCheck.append({
+                        label: catalog.i18nc("@label", "Top / Bottom"),
+                        colorId:  "layerview_skin"
+                    });
+                    typesLegenModelNoCheck.append({
+                        label: catalog.i18nc("@label", "Inner Wall"),
+                        colorId:  "layerview_inset_x"
+                    });
```
Comment (Ghostkeeper) [inline plugins/LayerView/LayerView.qml]: Why is this necessary? The `typesLegenModelNoCheck` is only initialised with these two settings, and never changed by anything afterwards. Clearing and re-appending its items should do nothing, right?

File: plugins/LayerView/LayerView.qml [hunk 2]
```diff
@@ -292,7 +303,7 @@ Item
                         color: UM.Theme.getColor(model.colorId)
                         border.width: UM.Theme.getSize("default_lining").width
                         border.color: UM.Theme.getColor("lining")
-                        visible: view_settings.show_legend
+                        visible: (UM.Preferences.getValue("layerview/layer_view_type") != 0) && view_settings.show_legend
```
Comment (Ghostkeeper) [inline plugins/LayerView/LayerView.qml]: This is now exactly the same as the parent, so this line has no effect any more. If the parent label is invisible, this rectangle will be too.

File: plugins/LayerView/LayerView.qml (hunk 1 above)
Comment (Ghostkeeper) [inline plugins/LayerView/LayerView.qml]: This `updateLegend` function is also called before the `componentCompleted`, causing these legend items to appear twice until you update for the first time.
doublelegenditems

**Comment by LipuFei** [comment]
@Ghostkeeper fixed.
